brainpy/_src/math/ndarray.py

```diff
@@ -1097,232 +1088,294 @@ def absolute_(self) -> 'Array':
     """
     return self.abs_()
```

Comment:
This will cause unexpected behavior. Instead of ``self.abs_()``, you should use ``self.abs()``. Because the former is performed in-pace.